### PR TITLE
Accept Provider in CompressionInputConfig

### PR DESCRIPTION
**Pull request by trivikr**
*Issue #, if available:*
Noticed while doing E2E testing in https://github.com/aws/aws-sdk-js-v3/pull/5625

*Description of changes:*
Accept Provider in CompressionInputConfig

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/middleware-compression/src/resolveCompressionConfig.ts [hunk 1]
```diff
@@ -1,19 +1,25 @@
+import { normalizeProvider } from "@smithy/util-middleware";
```
Comment (syall) [inline packages/middleware-compression/src/resolveCompressionConfig.ts:1]: The `@smithy/util-middleware` dependency needs to be added to `package.json`.

**Review by syall — CHANGES_REQUESTED** [review]

**Review by syall — APPROVED** [review]

File: packages/middleware-compression/src/resolveCompressionConfig.ts (hunk 1 above)
Comment (trivikr) [reply packages/middleware-compression/src/resolveCompressionConfig.ts:1]: Added in https://github.com/smithy-lang/smithy-typescript/commit/2a3d748440f4d586fa473bed3b7725bd004803bf

**Review by syall — APPROVED** [review]

File: packages/middleware-compression/src/compressionMiddleware.spec.ts [hunk 2]
```diff
@@ -15,8 +15,8 @@ describe(compressionMiddleware.name, () => {
   const mockBody = "body";
   const mockConfig = {
     bodyLengthChecker: jest.fn().mockReturnValue(mockBody.length),
-    disableRequestCompression: false,
-    requestMinCompressionSizeBytes: 0,
+    disableRequestCompression: () => Promise.resolve(false),
+    requestMinCompressionSizeBytes: () => Promise.resolve(0),
```
Comment (kuhe) [inline packages/middleware-compression/src/compressionMiddleware.spec.ts:19]:
```suggestion
requestMinCompressionSizeBytes: async () => 0,
```

really minor

File: packages/middleware-compression/src/compressionMiddleware.spec.ts [hunk 3]
```diff
@@ -15,8 +15,8 @@ describe(compressionMiddleware.name, () => {
   const mockBody = "body";
   const mockConfig = {
     bodyLengthChecker: jest.fn().mockReturnValue(mockBody.length),
-    disableRequestCompression: false,
-    requestMinCompressionSizeBytes: 0,
+    disableRequestCompression: () => Promise.resolve(false),
```
Comment (kuhe) [inline packages/middleware-compression/src/compressionMiddleware.spec.ts:18]:
```suggestion
    disableRequestCompression: async () => false,
```

minor

**Review by kuhe — APPROVED** [review]

File: packages/middleware-compression/src/compressionMiddleware.spec.ts (hunk 3 above)
Comment (trivikr) [reply packages/middleware-compression/src/compressionMiddleware.spec.ts]: Fixed in [`04f0594` (#1132)](https://github.com/smithy-lang/smithy-typescript/pull/1132/commits/04f0594d40f10c22acd7a7e3a0c452462ba80775)

File: packages/middleware-compression/src/compressionMiddleware.spec.ts (hunk 2 above)
Comment (trivikr) [reply packages/middleware-compression/src/compressionMiddleware.spec.ts]: Fixed in [`04f0594` (#1132)](https://github.com/smithy-lang/smithy-typescript/pull/1132/commits/04f0594d40f10c22acd7a7e3a0c452462ba80775)

**Pull request merged**